File: TypeScript/src/strategy-pattern/MallardDuck.ts
import Duck from './Duck';
import Quack from './Quack';
import FlyWithWings from './FlyWithWings';

export default class MallardDuck extends Duck {
  constructor() {
    super();
    this.quackBehavior = new Quack();
    this.flyBehavior = new FlyWithWings();
  }

  display() {
    console.log("I'am real Mallard duck");
  }
}

File: Ruby/strategy-pattern/mallard_duck.rb
# frozen_string_literal: true

require_relative './duck'
require_relative './quack'
require_relative './fly_with_wings'

# MallardDuck Class
class MallardDuck < Duck
  def initialize
    super
    @quack_behavior = Quack.new
    @fly_behavior = FlyWithWings.new
  end

  def display
    puts "I'am real Mallard duck"
  end
end
